planner/funcdep/only_full_group_by_test.go

```diff
@@ -0,0 +1,249 @@
+	tk.MustExec("create table t2(c int, d int);")
+	err = tk.ExecToErr("select t4.d from t1 left join (t2 as t3 join t2 as t4 on t4.d=3) on t1.a=10 group by \"\";")
+	require.NotNil(t, err)
+	require.Equal(t, err.Error(), "[planner:1055]Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'test.t4.d' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by")
+	tk.MustExec("select t4.d from t1 join (t2 as t3 left join t2 as t4 on t4.d=3) on t1.a=10 group by \"\";")
+	//tk.MustExec("drop table t1")
+	//tk.MustExec("drop view v1")
+	//tk.MustExec("create table t1(a int not null, b int)")
+	//tk.MustExec("create view v1 as select a as a, 2*a as b, coalesce(a,3) as c from t1")
+	//tk.MustExec("select v1.b from t1 left join v1 on 1 group by v1.a")
+
+	// test issue #25196
+	tk.MustExec("drop table if exists t1")
+	tk.MustExec("create table t1 (i1 integer, c1 integer);")
+	tk.MustExec("insert into t1 values (2, 41), (1, 42), (3, 43), (0, null);")
+	tk.MustExec("drop table if exists t2")
+	tk.MustExec("create table t2 (i2 integer, c2 integer, f2 float);")
+	tk.MustExec("insert into t2 values (0, 43, null), (1, null, 0.1), (3, 42, 0.01), (2, 73, 0.12), (null, 41, -0.1), (null, null, null);")
+	err = tk.ExecToErr("SELECT * FROM t2 AS _tmp_1 JOIN (SELECT max(_tmp_3.f2) AS _tmp_4,min(_tmp_3.i2) AS _tmp_5 FROM t2 AS _tmp_3 WHERE _tmp_3.f2>=_tmp_3.c2 GROUP BY _tmp_3.c2 ORDER BY _tmp_3.i2) AS _tmp_2 WHERE _tmp_2._tmp_5=100;")
+	require.NotNil(t, err)
+	require.Equal(t, err.Error(), "[planner:1055]Expression #3 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'test._tmp_3.i2' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by")
+
+	// test issue #22301 and #33056
+	tk.MustExec("drop table if exists t1, t2")
+	tk.MustExec("create table t1 (a int);")
+	tk.MustExec("create table t2 (a int, b int);")
+	tk.MustQuery("select t1.a from t1 join t2 on t2.a=t1.a group by t2.a having min(t2.b) > 0;")
+	tk.MustQuery("select t2.a, count(t2.b) from t1 join t2 using (a) where t1.a = 1;")
+	tk.MustQuery("select count(t2.b) from t1 join t2 using (a) order by t2.a;")
+
+	// test issue #30024
+	tk.MustExec("drop table if exists t1,t2;")
+	tk.MustExec("CREATE TABLE t1 (a INT, b INT, c INT DEFAULT 0);")
+	tk.MustExec("INSERT INTO t1 (a, b) VALUES (3,3), (2,2), (3,3), (2,2), (3,3), (4,4);")
+	tk.MustExec("CREATE TABLE t2 (a INT, b INT, c INT DEFAULT 0);")
+	tk.MustExec("INSERT INTO t2 (a, b) VALUES (3,3), (2,2), (3,3), (2,2), (3,3), (4,4);")
+	tk.MustQuery("SELECT t1.a FROM t1 GROUP BY t1.a HAVING t1.a IN (SELECT t2.a FROM t2 ORDER BY SUM(t1.b));")
```

Comment:
@time-and-fate This part is the case about the correlated agg resolving codes.

Reply:
Then we need some comments to clarify the code. It's not easy to understand the change. And it's not actually related to `only_full_group_by`.